Test/Unit/Helper/CreditmemoTest.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2023 Adyen NV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Test\Unit\Helper;
+
+use Adyen\Payment\Api\Data\OrderPaymentInterface;
+use Adyen\Payment\Helper\Data;
+use Adyen\Payment\Helper\Creditmemo;
+use Adyen\Payment\Model\CreditmemoFactory;
+use Adyen\Payment\Model\ResourceModel\Order\Payment;
+use Adyen\Payment\Test\Unit\AbstractAdyenTestCase;
+use Magento\Framework\App\Helper\Context;
+use Magento\Framework\Model\AbstractModel;
+
+class CreditmemoTest extends AbstractAdyenTestCase
+{
+    public function testCreateAdyenCreditMemo()
+    {
+        $pspReference = '12A34B56C78D910E';
+        $originalReference = 'E10D9C78B56A3412';
+        $refundAmount = 10.00;
+        $adyenOrderPaymentId = 5;
+
+        $paymentMock = $this->createConfiguredMock(\Magento\Sales\Model\Order\Payment::class, [
+            'getEntityId' => 5
+        ]);
+
+        $adyenOrderPaymentMock = $this->createConfiguredMock(OrderPaymentInterface::class, [
+            'getEntityId' => [OrderPaymentInterface::ENTITY_ID => $adyenOrderPaymentId]
+        ]);
+
+        $orderPaymentResourceModel = $this->createConfiguredMock(Payment::class, [
+            'getOrderPaymentDetails' => $adyenOrderPaymentMock->getEntityId()
```

Comment:
`getOrderPaymentDetails()` should return an array containing multiple `adyen_order_payment` rows not the `entity_id`.